canonical-web-and-design/maas-ui
# fix: use correct API request for commissioning without tests

## Done

- Fixed commissioning a machine without running tests
- Updated types so that only script names are used when running commissioning/testing

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the Tests tab of a machine that can be tested
- Open the Take action menu and click Commission
- Clear the testing script field so none are selected and submit the form
- Check that the tests tab does not update with a new test, and that the commissioning action was dispatched with `payload.params.extra.testing_scripts === ["none"]` instead of `[]`

## Fixes

Fixes #3955
